github/Requester.py

```diff
@@ -254,23 +254,34 @@ def __init__(self, login_or_token, password, base_url, timeout, client_id, clien
         self.__apiPreview = api_preview
         self.__verify = verify
 
-    def requestJsonAndCheck(self, verb, url, parameters=None, headers=None, input=None, cnx=None):
-        return self.__check(*self.requestJson(verb, url, parameters, headers, input, cnx))
+    def requestJsonAndCheck(self, verb, url, parameters=None, headers=None, input=None):
+        return self.__check(*self.requestJson(verb, url, parameters, headers, input, self.__customConnection(url)))
```

Comment:
we remove support for passing a magic-string as `cnx` variable for the outer method.
But for the inner method, all requests can now pass a custom connection as a parameter;

the method `__customConnection()` will parse the value of `url` and check if the request is meant for `https://api.github.com` or respective enterprise URL, which would mean it can re-use the existing API-connection and return `None`;
but if the URL is to another hostname or port than the existing connection, it will initiate new custom connection to the host and pass it as an argument for further processing.